finetune/eval/if_eval/start_end.py

```diff
@@ -0,0 +1,24 @@
+import random
+
+from finetune.eval.if_eval.rule import IFEvalRule, RuleId
+
+ENDS_WITH_PHRASES = [
+    "Let me know if you have additional questions.",
+    "Is there anything else I can help with?",
+    "Hope that helps.",
+]
+
+
+class EndsWithRule(IFEvalRule):
+    """Rule that enforces the response ends with an exact phrase."""
+
+    def __init__(self):
+        super().__init__(rule_id=RuleId.ENDS_WITH)
+        self.phrase = random.choice(ENDS_WITH_PHRASES)
+
+    def get_prompt(self, index: int = -1) -> str:
+        return f'End your response with the exact phrase "{self.phrase}"'
```

Comment:
We should probably include a parenthesis that says (do not include the quotations)

Reply:
IfEval does not do this. I think it is reasonable to exclude it and still expect the models to work. If anything this could help distinguish good vs bad ones.